cli/src/commands/download.ts

```diff
@@ -0,0 +1,119 @@
+import { CLIError } from '@oclif/errors';
+import * as path from 'path';
+import * as tar from 'tar';
+import fetch from 'node-fetch';
+import { spawn } from '@binaris/utils-subprocess';
+import Command from '../utils/command';
+import { Project } from '../utils/helpers';
+import flags from '../utils/cli-flags';
+
+export function statusNotOk(code: number): boolean {
+  return 200 > code || code >= 300;
+}
+
+export default class Download extends Command {
+  public static description = 'download application';
+  public static examples = [
+    `$ ${Command.cliBinName} download 123456`,
+  ];
+
+  public static args = [
+    {
+      name: 'ID',
+      required: true,
+      description: 'application id',
+    },
+  ];
+
+  public static flags = {
+    ...Command.flags,
+    verbose: flags.boolean({
+      char: 'v',
+      description: 'Be verbose',
+      default: false,
+    }),
+  };
+
+  public static strict = true;
+
+  public async run() {
+    const {
+      args : { ID: applicationId },
+      flags: { verbose },
+    } = this.parse(Download);
+    await this.authenticate();
+    const applications = await this.lycanClient.listApps();
+    const application = applications.find((app) => app.id === applicationId);
+    if (!application) {
+      if (verbose) {
+        this.log('All applications:', applications);
```

Comment:
We really need a proper log, and then always log and just change disposition of level `debug` / `verbose`.  Not in this PR.

Reply:
totally agree!
It should also affect all commands